### PR TITLE
Fix application id issue

### PR DESCRIPTION
**Pull request by tyrannosaurus-becks**
A customer was receiving the following output on version 2.3 of PCF Dev:
```
Service broker error: failed to create mounts /cf//secret=generic, /cf//transit=transit: Error making API request. URL: POST https://gmconfig-vault-test-q3mi.gm.com:8200/v1/sys/mounts/cf/transit Code: 400. Errors: * existing mount at cf/transit/
```
This code adds a test duplicating that error, and then fixing it. At first, the output from the new test is:
```
=== RUN   TestBroker_Bind_Unbind_No_Application_ID
[INFO] binding service binding-id to instance instance-id
[DEBUG] looking up instance instance-id from cache
[DEBUG] creating mounts /cf//secret=generic, /cf//transit=transit
[ERR] failed to create mounts /cf//secret=generic, /cf//transit=transit: Error making API request.  URL: POST http://127.0.0.1:46615/v1/sys/mounts/cf/secret Code: 400. Errors:  
--- FAIL: TestBroker_Bind_Unbind_No_Application_ID (0.00s)
    broker_test.go:79: failed to create mounts /cf//secret=generic, /cf//transit=transit: Error making API request.
        
        URL: POST http://127.0.0.1:46615/v1/sys/mounts/cf/secret
        Code: 400. Errors:
        
        
FAIL
```
However, with the application-level storage not being issued if no application ID is provided, the test passes.

This code has been successfully tested with the following cf development environments:
```
$ cf dev version  
CLI: 0.0.15
BUILD: 54 (35368e4)

cf: v7.9.0
cf-mysql: 36.19.0

$ cf -v           
cf version 6.41.0+dd4c76cdd.2018-11-28
```

### CODE REVIEW

File: broker_test.go [hunk 1]
```diff
@@ -113,6 +113,55 @@ func TestBroker_Bind_Unbind(t *testing.T) {
...
+	shared, ok := credMap["backends_shared"]
+	if !ok {
+		t.Fatalf("expected backends_shared but they're not in %+v", credMap)
+	}
+	sharedMap, ok := shared.(map[string]interface{})
+	if !ok {
+		t.Fatalf("expected a backends_shared map but received %+v", shared)
+	}
+	if sharedMap["organization"] != "cf/organization-guid/secret" {
+		t.Fatalf("expected cf/space-guid/secret but received %s", sharedMap["organization"])
```
Comment (kalafut) [inline broker_test.go]:
```suggestion
		t.Fatalf("expected cf/organization-guid/secret but received %s", sharedMap["organization"])
```

File: vault_test.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+		ApplicationGUID: "application-id",
+	}
+	if err := GeneratePolicy(w, info); err != nil {
+		t.Fatal(err)
+	}
+	result := w.String()
+	if result != expected {
+		t.Fatalf("received unexpected policy of %s", result)
+	}
+}
```
Comment (kalafut) [inline vault_test.go]: Probably good to test the case with `ApplicationGUID == ""`.

Comment (tyrannosaurus-becks) [reply vault_test.go]: Yes, good point. Added!

**Comment by tyrannosaurus-becks** [comment]
@kalafut thank you!

**Pull request merged**